api/controllers/send_api_controller.rb

```diff
@@ -84,7 +85,11 @@
 
       # If we're not authenticated, don't continue
       if authenticated_domain.nil?
-        error 'UnauthenticatedFromAddress'
+        if mail.from.nil? || mail.from.empty? || !/@/.match(mail.from.first)
```

Comment:
I would suggest removing the regex check. The from address is not "missing" if it's merely invalid. Also, I assume we have Rails extensions and could therefore just use `mail.from.blank?`